File: Easy/BinaryInorderTraversal_94.py
from typing import List, Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class BinaryInorderTraversal_94:
    def __init__(self):
        self.result = []  # Use an instance variable instead of a class variable

    def inOrder(self, root: Optional[TreeNode]):
        if root is None:
            return
        self.inOrder(root.left)
        self.result.append(root.val)
        self.inOrder(root.right)

    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.result = []  # Reset result for each traversal
        self.inOrder(root)
        return self.result

# Example Usage:
# Constructing a sample binary tree
#         1
#          \
#           2
#          /
#         3

node = TreeNode(1)
node.right = TreeNode(2)
node.right.left = TreeNode(3)

obj = BinaryInorderTraversal_94()
print(obj.inorderTraversal(node))  # Output: [1, 3, 2]

File: Easy/BinaryInorderTraversal_94.cpp
#include <iostream>
#include <vector>
class TreeNode
{
public:
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};
class BinaryInorderTraversal_94
{
public:
    std::vector<int> inorderTraversal(TreeNode *root)
    {
        return {};
    }
};
int main()
{
    TreeNode *root;
    root = new TreeNode(1);
    root->right = new TreeNode(2);
    root->right->left = new TreeNode(3);
    BinaryInorderTraversal_94 b1;
    b1.inorderTraversal(root);
    return 0;
}